src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/PackageManagerUICommandHandler.cs

```diff
@@ -272,50 +272,44 @@ private async Task<vsShellInterop.IVsWindowFrame> CreateNewWindowFrameAsync(stri
+            }
 
-            var uiController = UIFactory.Value.Create();
-
-            var model = new PackageManagerModel(
-                uiController,
-                isSolution: false,
-                editorFactoryGuid: GuidList.NuGetEditorType);
-
-            var vsWindowSearchHostfactory = await _asyncServiceProvider.GetServiceAsync(typeof(vsShellInterop.SVsWindowSearchHostFactory)) as vsShellInterop.IVsWindowSearchHostFactory;
-            var vsShell = await _asyncServiceProvider.GetServiceAsync(typeof(vsShellInterop.SVsShell)) as vsShellInterop.IVsShell4;
+            NuGetProject nugetProject = new NuGetProjectInternal(projectGuid);
+            var uiController = UIFactory.Value.Create(nugetProject);
+            var model = new PackageManagerModel(uiController, isSolution: false, editorFactoryGuid: GuidList.NuGetEditorType);
+            var vsWindowSearchHostfactory = await _asyncServiceProvider.GetServiceAsync(typeof(SVsWindowSearchHostFactory)) as IVsWindowSearchHostFactory;
+            var vsShell = await _asyncServiceProvider.GetServiceAsync(typeof(SVsShell)) as IVsShell4;
             var control = new PackageManagerControl(model, Settings.Value, vsWindowSearchHostfactory, vsShell, OutputConsoleLogger.Value);
+            var caption = string.Format(CultureInfo.CurrentCulture, Resx.Label_NuGetWindowCaption, Path.GetFileNameWithoutExtension(workspaceVisualNodeBase.NodeMoniker));
 
-            var caption = string.Format(
-                CultureInfo.CurrentCulture,
-                Resx.Label_NuGetWindowCaption,
-                Path.GetFileNameWithoutExtension(projectPath));
-
-            vsShellInterop.IVsWindowFrame windowFrame;
-            var uiShell = await _asyncServiceProvider.GetServiceAsync(typeof(vsShellInterop.SVsUIShell)) as vsShellInterop.IVsUIShell;
+            var uiShell = await _asyncServiceProvider.GetServiceAsync(typeof(SVsUIShell)) as IVsUIShell;
             Assumes.Present(uiShell);
-            var hr = 0;
-            int[] pfDefaultPosition = null;
 
+            int[] pfDefaultPosition = null;
+            IVsWindowFrame windowFrame;
             var windowPane = new PackageManagerToolWindowPane(control);
-            hr = uiShell.CreateToolWindow(
-                (uint)__VSCREATETOOLWIN.CTW_fInitNew,
-                0,              // dwToolWindowId - singleInstance = 0
-                windowPane,     // ToolWindowPane
-                Guid.Empty,     // rclsidTool = GUID_NULL
-                Guid.NewGuid(), // TODO: should be projectGuid...so persistance info works: https://github.com/NuGet/Home/issues/9636
-                Guid.Empty,     // reserved - do not use - GUID_NULL
-                null,           // IServiceProvider
-                caption,
-                pfDefaultPosition,
-                out windowFrame);
+            ErrorHandler.ThrowOnFailure(uiShell.CreateToolWindow(
+                                            (uint)__VSCREATETOOLWIN.CTW_fInitNew,
```

Comment:
nit: seems like a lot of space

Reply:
Its indented under the most immediate object calling the method, is that not the norm in NuGet source?  I can move it to indent under ErrorHandler but when I look at it at first glance it seems out of place there.

Reply:
Maybe break up the line a little more:

```cs
ErrorHandler.ThrowOnFailure(
    uiShell.CreateToolWindow(
        (uint)__VSCREATETOOLWIN.CTW_fInitNew,
        ...));
```